app/ui-react/syndesis/src/modules/connections/components/Connections.tsx

```diff
@@ -78,16 +79,11 @@ export class Connections extends React.Component<IConnectionsProps> {
                               };
 
                               const configurationRequired =
```

Comment:
Because of [this](https://github.com/syndesisio/syndesis-react/pull/409/files#diff-c8c891bfe81f60cb81839bf4031cb8c7R19) you don't need this anymore, `c.configRequired` should hold your value.

Reply:
>I agree with Stan, we either do a full refactoring to smooth all the raw edges, or not.
>I'm ok with continuing working on this 'til everything is DRYed up and makes sense.

As long as you're ok with questions and helping to explain, I'm always happy to refactor.

`configRequired` is not a property of `ConnectionOverview`, will run into this for `isTechPreview` as well, I'm guessing--what is the best approach here? Should I make an interface that extends `ConnectionOverview`?

Reply:
I think we should be fine without a new interface. When you get connections out from `useConnections` (or `WithConnections`) the object that's returned is an object that does respect the `Connection` interface, but it's also extended with a number of properties that the app needs.

These connection objects are then eventually passed to the `toUIStep` function to convert them, and all their properties are spread in the new `IUIStep` one.
So if we simply extend the `IUIStep` interface to have `configRequired` and `isTechPreview` defined, we should be good.

Reply:
I'm able to do this for `EditorSteps.tsx`, for instance, but not for `Connections.tsx` since it seems to never use `IUIStep`, where in `IConnectionsProps`: `connections: ConnectionOverview[];`. Should that instead be `IUIStep` or have I missed something?

Reply:
`Connetions.tsx` will use connections coming directly from the API, so already transformed. `c.configRequired` will be available then.

Reply:
But I see now that we might need to create an extended `IConnection` interface, since `Connections.tsx`. Boring, but yeah, needed.